pose_prediction/plugin.cpp

```diff
@@ -99,9 +156,203 @@ class pose_prediction_impl : public pose_prediction {
 		Eigen::Quaternionf raw_o (pose.orientation.w(), -pose.orientation.y(), pose.orientation.z(), -pose.orientation.x());
 
 		swapped_pose.orientation = apply_offset(raw_o);
+        swapped_pose.sensor_time = pose.sensor_time;
 
         return swapped_pose;
     }
+
+    // Slightly modified copy of OpenVINS method found in propagator.cpp
+    // Returns a pair of the predictor state_plus and the time associated with the
+    // most recent imu reading used to perform this prediction.
```

Comment:
@mhuzai If we are modifying a function from OpenVINS should we do it actually in OpenVINS?

Reply:
In this case I dont think so because this change make this method read switchboard plugs which we obviously dont want to be doing in OpenVINS